pci/src/vfio.rs

```diff
@@ -48,10 +48,10 @@ pub(crate) const VFIO_COMMON_ID: &str = "vfio_common";
 pub enum VfioPciError {
     #[error("Failed to create user memory region: {0}")]
     CreateUserMemoryRegion(#[source] HypervisorVmError),
-    #[error("Failed to DMA map: {0}")]
-    DmaMap(#[source] vfio_ioctls::VfioError),
-    #[error("Failed to DMA unmap: {0}")]
-    DmaUnmap(#[source] vfio_ioctls::VfioError),
+    #[error("{1}: Failed to DMA map: {0}")]
+    DmaMap(#[source] vfio_ioctls::VfioError, String),
```

Comment:
Does it not work to use the PciBdf here directly?  It implements `Display.

```suggestion
    DmaMap(#[source] vfio_ioctls::VfioError, PciBdf),
```

That way it would be more inspectable in code as well.

Reply:
> Does it not work to use the PciBdf here directly? It implements Display.
> 
> That way it would be more inspectable in code as well.

Unfortunately, it doesn't work as expected. The output seems to be `PciBdf(24)`, instead of something like `"0000:00:03.0"`.

Reply:
That's very strange — the `Display` implementation for `PciBdf` definitely returns the latter format, and the thiserror documentation suggests it should use `Display` unless you use `:?` in the format string…

Reply:
@arv-crusoe @alyssais If it's printing `PciBdf(24)` then that is almost certainly because the error itself is being printed using `"{:?}"` rather than `"{}"` - unfortunately we are still eradicating those.

Reply:
The correct fix is to use `PciBdf` as @alyssais  suggests and trace back through the error path to find where it's being printed out using `Debug` when it should be `Display`.

Reply:
I looked into this a little more and the issue turns out to be that the layer above this, DeviceManagerError does not implement the Display function. So it is propagating the Debug output to the caller.
The offending code is in  device_manager.rs
```
#[error("Error from device manager: {0:?}")]
    DeviceManager(DeviceManagerError),
```
It seems like a good idea to make the change to allow propagating the lower level error here, but it may be too much scope creep to add the Display for all the different errors that the DeviceManagerError reports.

Reply:
Yeah, `DeviceManagerError` is a big one

Reply:
Would you like me to take a stab at creating a patch with some of the DeviceManagerError arms filled out? That would at least start the process of fleshing out some of the error messages or would you prefer to keep the patch as-is and then have the rest of the changes for the `DeviceManagerError` plus the PciBdf changes in a separate patch?

Reply:
I have added the following patch for the rest of the changes:
https://github.com/cloud-hypervisor/cloud-hypervisor/pull/6908
Lmk if this is sufficient.

Reply:
Just implement `fmt::Debug` on `PciBdf` to print the address like the `Display` method. This will then get you pretty printing of the PCI address in the error messages.

Reply:
Done. Please review the changes. Do you want me to rebase the changes into a single commit?

Reply:
I would appreciate thatl. I think this should logically be:
- A commit to implement `fmt::Debug` for `PciBdf`
- Any changes to make the most use of it.

Reply:
I've re-ordered the commits. Please let me know if this works.